src/muse/agents/agent.py

```diff
@@ -263,22 +294,75 @@ def next(
         """
         from logging import getLogger
 
-        # dataset with intermediate computational results from search
-        # makes it easier to pass intermediate results to functions, as well as
-        # filter them when inside a function
+        current_year = self.year
+
+        # Skip forward if demand is zero
         if demand.size == 0 or demand.sum() < 1e-12:
             self.year += time_period
             return None
 
+        # Calculate the search space
         search_space = (
             self.search_rules(self, demand, technologies, market).fillna(0).astype(int)
         )
 
+        # Skip forward if the search space is empty
         if any(u == 0 for u in search_space.shape):
             getLogger(__name__).critical("Search space is empty")
```

Comment:
In theory it can continue, the agent just won't perform any investments for that year. If this happens, then it probably means there's a mistake with the way the model is defined, so this is something the user should definitely pay attention to. I'm a bit concerned that a lot a important messages are getting ignored, so have suggested these get outputted to a file (#498)